File: core_arch/src/main/java/com/huyingbao/core/arch/scope/FragmentScope.java
package com.huyingbao.core.arch.scope;

import java.lang.annotation.Retention;

import javax.inject.Scope;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * FragmentScope创建出来的单例保留在FragmentComponent实例中
 *
 * @author liujunfeng
 * @date 2019/1/1
 */
@Scope
@Retention(RUNTIME)
public @interface FragmentScope {
}

File: core-arch/src/main/java/com/huyingbao/core/arch/scope/FragmentScope.kt
package com.huyingbao.core.arch.scope

import javax.inject.Scope

/**
 * [FragmentScope]创建出来的单例保留在FragmentComponent实例中
 *
 * Created by liujunfeng on 2019/1/1.
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class FragmentScope
